src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java

```diff
@@ -89,32 +89,33 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
                             refIds.size());
                     }
                 }
+            }
 
-                if (numFilesSeen != uniqueRefIds.size()) {
-                    logger.info("some files are repeated. {} unique values across {} total file references",
-                        uniqueRefIds.size(), numFilesSeen);
-                }
-                List<String> uniqueRefIdsAsList = new ArrayList<>(uniqueRefIds);
-                Dataset dataset = datasetService.retrieve(snapshotSource.getDataset().getId());
-
-                String addFilesTimer = performanceLogger.timerStart();
-                fileDao.addFilesToSnapshot(dataset, snapshot, uniqueRefIdsAsList);
-                performanceLogger.timerEndAndLog(
-                    addFilesTimer,
-                    context.getFlightId(),
-                    this.getClass().getName(),
-                    "fileDao.addFilesToSnapshot",
-                    uniqueRefIds.size());
-
-                String addDependenciesTimer = performanceLogger.timerStart();
-                dependencyDao.storeSnapshotFileDependencies(dataset, snapshot.getId().toString(), uniqueRefIdsAsList);
-                performanceLogger.timerEndAndLog(
-                    addDependenciesTimer,
-                    context.getFlightId(),
-                    this.getClass().getName(),
-                    "dependencyDao.storeSnapshotFileDependencies",
-                    uniqueRefIds.size());
+            if (numFilesSeen != uniqueRefIds.size()) {
+                logger.info("some files are repeated. {} unique values across {} total file references",
+                    uniqueRefIds.size(), numFilesSeen);
             }
+
+            List<String> uniqueRefIdsAsList = new ArrayList<>(uniqueRefIds);
+            Dataset dataset = datasetService.retrieve(snapshotSource.getDataset().getId());
+
+            String addFilesTimer = performanceLogger.timerStart();
+            fileDao.addFilesToSnapshot(dataset, snapshot, uniqueRefIdsAsList);
+            performanceLogger.timerEndAndLog(
```

Comment:
I hadn't seen the perf API before and this seems a bit messy and makes the code harder to read. Now that we have a few uses of it, it might be worth revisiting and trying to make a cleaner API.